File: csharp/main/LiteralFloat32.cs
using System;

namespace Deltix.Luminary
{
	public sealed class LiteralFloat32 : Literal, ILiteral<Single>
	{
		public LiteralFloat32(Single value)
			: base(LiteralKind.Float32)
		{
			Value = value;
		}

		public Single Value { get; }

		public override String ToString()
		{
			if (Double.IsNaN(Value))
				return "Float32.NaN";
			if (Double.IsPositiveInfinity(Value))
				return "Float32.POSITIVE_INFINITY";
			if (Double.IsNegativeInfinity(Value))
				return "Float32.NEGATIVE_INFINITY";
			return $"{Value:G9}f32";
		}
	}
}

File: java/src/main/java/deltix/luminary/LiteralFloat32.java
package deltix.luminary;

public final class LiteralFloat32 extends Literal {
    private final float value;

    public LiteralFloat32(float value) {
        super(LiteralKind.FLOAT32);
        this.value = value;
    }

    public float getValue() {
        return value;
    }

    @Override
    public String toString() {
        if (Double.isNaN(value))
            return "Float32.NaN";
        if (Double.isInfinite(value))
            return value > 0 ? "Float32.POSITIVE_INFINITY" : "Float32.NEGATIVE_INFINITY";
        return Float.toString(value) + "f32";
    }
}
